### PR TITLE
Support for geo: prefix

### PR DESCRIPTION
**Pull request by m-kuhn**
Fixes https://github.com/opengisch/QField/issues/3782

### CODE REVIEW

**Comment by nirvn** [comment]
@m-kuhn , interesting way to go at it :) I was thinking of adding a parser in the QR code reader that would remove geo: when matching a regex along the lines of "geo:\s*\d\s+\d" but this works too. And we keep the source QR code string, which is educative :)

File: src/core/locator/gotolocatorfilter.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ void GotoLocatorFilter::fetchResults( const QString &string, const QgsLocatorCon
   if ( !match.hasMatch() || !firstOk || !secondOk )
   {
     // Digit detection using user locale failed, use default C decimal separators
-    separatorRx = QRegularExpression( QStringLiteral( "^([0-9\\-\\.]*)[\\s\\,]*([0-9\\-\\.]*)$" ) );
+    separatorRx = QRegularExpression( QStringLiteral( "^(?:(?i)geo:)([0-9\\-\\.]*)[\\s\\,]*([0-9\\-\\.]*)$" ) );
```
Comment (nirvn) [inline src/core/locator/gotolocatorfilter.cpp:68]:
```suggestion
    separatorRx = QRegularExpression( QStringLiteral( "^(?:(?i)geo:){0,1}([0-9\\-\\.]*)[\\s\\,]*([0-9\\-\\.]*)$" ) );
```

**Review by nirvn — APPROVED** [review]

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a864228a68b3f2fc3e26d20c9daaf20dc80b4c53: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3785-24fe374f484460e2f02cce75403ced2fadbb608f-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3785-24fe374f484460e2f02cce75403ced2fadbb608f-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3785-24fe374f484460e2f02cce75403ced2fadbb608f-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3785-24fe374f484460e2f02cce75403ced2fadbb608f-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3785-24fe374f484460e2f02cce75403ced2fadbb608f-arm64-android.apk)
- [all access arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3785-24fe374f484460e2f02cce75403ced2fadbb608f-arm-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3785-24fe374f484460e2f02cce75403ced2fadbb608f-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-3785-24fe374f484460e2f02cce75403ced2fadbb608f-x86-android.apk)
</details>

**Comment by m-kuhn** [comment]
Would you then have produced a QgsPoint? Or just eliminated the prefix?

**Comment by nirvn** [comment]
@m-kuhn , I would have just eliminated the prefix. 

A `processQrCodeString( const QString &str )` function could also have taken care of other simple prefix (eg `mailto:`).

**Comment by m-kuhn** [comment]
I like the current way of keeping information 😛

**Comment by m-kuhn** [comment]
Tested, works

**Comment by nirvn** [comment]
@m-kuhn , I do too. 

Re CRS, I don't know, we can leave it as is for people to abuse and encode geo:X,Y in a different CRS :)

**Comment by m-kuhn** [comment]
> @m-kuhn , I do too.
> 
> Re CRS, I don't know, we can leave it as is for people to abuse and encode geo:X,Y in a different CRS :)

They shouldn't. If they wany want to, they should create a format extension that includes the CRS. I.e. I'd prefer to actively avoid this to not incentivising such an abuse.

**Pull request merged**